opencsp/common/lib/render/figure_management.py

```diff
@@ -517,78 +517,36 @@ def print_figure_summary() -> None:
-        - the calling code be executed as a subprocess
-        - the subprocess is terminated when save_all returns
-
     Args:
     -----
         - output_path (str): The directory to save figures to.
         - format (str): The file format for figures. None for RenderControlFigureRecord.save default. Defaults to None.
-        - timeout (float): How many seconds to wait for the image to be saved. None to wait indefinitely. Defaults to None. FIXME make this work on windows.
-        - raise_on_timeout (bool): Whether to raise an exception if timeout elapse. Defaults to False.
 
     Returns:
     --------
         - figs: list[str] The list of image files
         - txts: list[str] The list of image descriptor text files
-        - failed: list[RenderControlFigureRecord] The list of figure records that failed to save (only returned if timeout is not None)
+        - failed: list[RenderControlFigureRecord] The list of figure records that failed to save
     """
-    global fig_record_list
+    global fig_record_list  # TODO: convert from global to class member or save_all_figures parameter
     figs: list[str] = []
     txts: list[str] = []
     failed: list[RenderControlFigureRecord] = []
 
-    if timeout == None:
-        for fig_record in fig_record_list:
-            t1, t2 = fig_record.save(output_path, format=format, close_after_save=False)
-            figs.append(t1)
-            txts.append(t2)
-
-        return figs, txts
-    else:
-        # Save each figure with a timeout on how long to wait for the figure to be saved.
-        from threading import Thread
-
+    try:
         for fig_record in fig_record_list:
-            # start the save
-            results = []
-            t = Thread(
-                target=lambda: results.append(fig_record.save(output_path, format=format, close_after_save=False))
-            )
-            t.start()
-
-            # wait for the save to finish
-            t.join(timeout)
-            if not t.is_alive():
-                # join the thread again, in case it finished between the timeout and the is_alive()
-                t.join(0.1)
-                # done saving the figure
-                # get the results and register the success
-                t1, t2 = results[0]
-                figs.append(t1)
-                txts.append(t2)
-                break
-
-            # the figure failed to save before the timeout
-            err_msg = f"Error: figure_management.save_all_figures: failed to save figure {fig_record.figure_num} \"{fig_record.name}\""
-            if raise_on_timeout:
-                lt.error_and_raise(RuntimeError, err_msg)
-            else:
-                lt.error(err_msg)
-            failed.append(fig_record)
-
-        return figs, txts, failed
+            fig_file, txt_file = fig_record.save(output_path, format=format, close_after_save=False)
+            figs.append(fig_file)
+            txts.append(txt_file)
+    except RuntimeError:

```

Comment:
I'm not usually in the habit of catching all exceptions, but this may be the time to do so. All exceptions that are safe to intercept without the program state becoming invalidated fall under the umbrella of "Exception". Consider changing this to something like:

```python
except Exception as ex:
    err_msg = "Error in figure_management.save_all_figures(): " +
              f"failed to save figure {fig_record.figure_num} \"{fig_record.name}\" due to exception \"{repr(ex)}\""
```